### PR TITLE
5 to 6 (main)

### PR DESCRIPTION
**Pull request by adlarkin**
# ➡️ Forward port

Port `ign-gazebo5` to `main`

Branch comparison: https://github.com/ignitionrobotics/ign-gazebo/compare/main...ign-gazebo5

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

### CODE REVIEW

File: examples/plugin/hello_world/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+find_package(ignition-cmake2 REQUIRED)
+
+project(Hello_world)
+
+ign_find_package(ignition-plugin1 REQUIRED COMPONENTS register)
+set(IGN_PLUGIN_VER ${ignition-plugin1_VERSION_MAJOR})
+
+ign_find_package(ignition-gazebo5 REQUIRED)
+set(IGN_GAZEBO_VER ${ignition-gazebo5_VERSION_MAJOR})
```
Comment (chapulina) [inline examples/plugin/hello_world/CMakeLists.txt:11]:
```suggestion
set(IGN_GAZEBO_VER ${ignition-gazebo6_VERSION_MAJOR})
```

File: src/gui/plugins/banana_for_scale/BananaForScale.cc [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  ignition::gui::App()->sendEvent(
+      ignition::gui::App()->findChild<ignition::gui::MainWindow *>(),
+      &event);
+
+  IGN_UTILS_WARN_IGNORE__DEPRECATED_DECLARATION
+  ignition::gazebo::gui::events::SpawnPreviewPath oldEvent(sdfPath);
+  ignition::gui::App()->sendEvent(
+      ignition::gui::App()->findChild<ignition::gui::MainWindow *>(),
+      &oldEvent);
+  IGN_UTILS_WARN_RESUME__DEPRECATED_DECLARATION
```
Comment (chapulina) [inline src/gui/plugins/banana_for_scale/BananaForScale.cc:115]: I haven't checked CI, but I think this needs to be removed to compile

**Comment by adlarkin** [comment]
It looks like CI is not picking up the changes from https://github.com/ignitionrobotics/ign-gui/pull/258 ([here's the CI failure](https://build.osrfoundation.org/job/ignition_gazebo-ci-pr_any-ubuntu_auto-amd64/6650/consoleFull#-13308563306ea37b8d-a6d4-40ae-8431-d90c018842af), indicating that the `gui::Application` class does not have the changes made in https://github.com/ignitionrobotics/ign-gui/pull/249). I thought that the nightlies were used for CI on unreleased branches, so the nightly packages of `ign-gui` should be up-to-date by now since the gui forward port I am referencing here was merged just over a week ago... am I misunderstanding something?

**Comment by chapulina** [comment]
> It looks like CI is not picking up the changes

@adlarkin , see https://github.com/ignition-release/ign-gui6-release/pull/2

**Review by chapulina — APPROVED** [review]
LGTM with happy CI

**Comment by adlarkin** [comment]
@osrf-jenkins retest this please

**Pull request merged**